Documentation/project-docs/platform-names.md

```diff
@@ -0,0 +1,61 @@
+
+.NET Native is a compiler tool chain that will produce native code ahead-of-time (AOT), as opposed to just-in-time (JIT). The compilation can happen on the developer machine as well as on the store side, which allows blending AOT with the benefits of servicing.
+
+You can think of .NET Native as an evolution of NGEN: NGEN basically simply runs the JIT up front, the code quality and behavior is identical to the JITed version. Another downside of NGEN is that it happens on the user's machine, rather than the developer's machine. NGEN is also at the module level, i.e. for each MSIL assembly there is a corresponding NEGEN'ed assembly that contains the native code. .NET Native on the other hand is a C++ like compiler and linker. It will remove unused code, spend more time optimizing it, and produce a single, merged module that represents the closure of the application.
+
+UWP was the first application model that was supported by .NET Native. We now also support building native console applications for Windows, Mac and Linux.
+
+## Rotor
+
+**Also referred to as**: Shared Source Common Language Infrastructure (SSCLI)
+
+Pretty much at the same time the .NET Framework was released, Microsoft also published Rotor, which is the source code for an implementation of ECMA 335 (Common Language Infrastructure), which is the specification behind .NET.
+
+While parts of the source were identical with the .NET Framework, many pieces had prototypic implementations instead: the purpose of Rotor wasn't to provide a production ready .NET implementation but to provide a platform for research, academia, and validation that the ECMA 335 specification itself can be implemented.
+
+It's also worth pointing out that the source code of Rotor was not an open source license (i.e. not approved by OSI).
+
+## Mono
+
+Mono is an open source alternative to the .NET Framework. Mono started around the same time the .NET Framework was first released. Since Microsoft didn't release Rotor as open source, Mono was forced to start from scratch and is thus a complete re-implementation of the .NET Framework with no shared code.
+
+When .NET Core was released under the MIT license, Microsoft also released large chunks of the .NET Framework under the MIT license as well, which can be found [here](https://github.com/microsoft/referencesource). This enabled the Mono community to use the same code the .NET Framework uses in order to close gaps and avoid behavioral differences.
+
+## Xamarin
+
+Xamarin is a commercial offering for building mobile applications targeting Android, iOS and MacOS . It's based on Mono, and surfaces a different API profile, called the mobile profile. The subsetting was necessary to reduce the footprint, both by shipping smaller versions of the system libraries as well as making them more linker friendly.   Xamarin ships a full static compiler on iOS and PlayStation4, as the platform does not support dynamic code generation.
```

Comment:
Not sure if worth mentioning, but they ship static compiler for Android as well and the developer decides whether to use the AOT or JIT compiler.